### DISCUSSION TITLE
Why am I seeing `set` run immediately after `get` in different pods?

### DISCUSSION THREAD

**Discussion by dross15** [caching-tools/next-shared-cache#647] [Q&A]
Hi! I am running a Next.js app with EKS and using s3 as my shared cache. The first time I hit a page I see X-Nextjs-Cache as MISS, which makes sense, and if I add logs to my cache-handler, I see that the get attempt fails and then it calls set which is all expected behavior. However, if I reload the page, and this time the traffic goes through a different pod, I see the get method succeed, see X-Nextjs-Cache as STALE (this is a minute or two later, so it shouldn't be stale yet), and then shortly after I see the set method fired from within that same pod even though the get method has just succeeded. Does anyone know why this might be the case? Thanks! cc @better-salmon

**Comment by abudriman** [comment]
any updates ? I also experience this with ` "@neshca/cache-handler": "1.3.1"`. im running nextjs in cluster mode with `pm2` , somehow the cache always  STALE on the first hit for each different cluster instances

[edit]
just found out it is expected behavior from next.js.
